src/panoptes/pocs/focuser/focuser.py

```diff
@@ -401,51 +393,57 @@ def _autofocus(self,
             focus_range = focus_range[0]
             focus_step = focus_step[0]
 
+        # Get focus steps.
         focus_positions = np.arange(max(initial_focus - focus_range / 2, self.min_position),
                                     min(initial_focus + focus_range / 2, self.max_position) + 1,
                                     focus_step, dtype=np.int)
         n_positions = len(focus_positions)
 
-        thumbnails = np.zeros((n_positions, thumbnail_size, thumbnail_size),
-                              dtype=initial_thumbnail.dtype)
+        # Set up empty array holders
+        thumbnails = np.zeros((n_positions, thumbnail_size, thumbnail_size), dtype=initial_thumbnail.dtype)
```

Comment:
Do we need to keep all the thumbnails in memory? Can we just keep the one with the best focus metric?